packages/dashboard/src/views/Trade.js

```diff
@@ -0,0 +1,246 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { useSelector, useDispatch } from 'react-redux'
+import { Container, Row, Col } from 'shards-react'
+import moment from 'moment'
+
+import { tradeActions } from '../redux/actions'
+
+import MainChart from '../components/MainChart'
+import SecurityList from '../components/SecurityList'
+
+import PageTitle from './../components/common/PageTitle'
+import NewDraft from './../components/blog/NewDraft'
+import Discussions from './../components/blog/Discussions'
+
+const Trade = ({ chartData }) => {
+  // state selection
+  const selectedAccount = useSelector(state => state.trade.selectedAccount)
+  const accounts = useSelector(state => state.trade.accounts)
+  const selectedSecurity = useSelector(state => state.trade.selectedSecurity)
+
+  // dispatching
+  const dispatch = useDispatch()
+
+  // select security from portfolio
+  const selectSecurity = security => {
+    setSelectedRange('1d')
+    dispatch(tradeActions.selectSecurity(security))
+  }
+
+  // internal state
+  const [selectedRange, setSelectedRange] = useState('1d')
+  const [currentChartData, setCurrentChartData] = useState(chartData)
+
+  // useEffects
+  useEffect(() => {
+    // if we have data for a selected range, generate new chart data
+    if (!!selectedSecurity?.historicQuotes?.[selectedRange]) {
```

Comment:
didnt know you could do this with the question marks. pretty clutch

Reply:
Optional chaining, it is clutch. New feature in ES7 that isn't even in regular JS yet. I'd argue to access every object property like that, just so we don't break our app on weird cases.